drivers/gpu/drm/rp1/rp1-dpi/rp1_dpi.c

```diff
@@ -379,21 +392,26 @@ static int rp1dpi_platform_probe(struct platform_device *pdev)
 			break;
 
 		if (!of_property_read_string(np1, "function", &str) && !strcmp(str, "dpi")) {
-			for (j = 0; !dpi->gpio1_used; j++) {
+			for (j = 0; ret < 0xE; j++) {
 				if (of_property_read_string_index(np1, "pins", j, &str))
 					break;
 				if (!strcmp(str, "gpio1"))
-					dpi->gpio1_used = true;
+					ret |= BIT(1);
+				else if (!strcmp(str, "gpio2"))
+					ret |= BIT(2);
+				else if (!strcmp(str, "gpio3"))
+					ret |= BIT(3);
 			}
-			for (j = 0; !dpi->gpio1_used; j++) {
+			for (j = 0; ret < 3; j++) {
```

Comment:
This condition looks wrong - shouldn't it also be 0xE (although the capital bugs me)?

Reply:
Ah yes, thanks. You can see how the code evolved from checking only GPIO1 to 2 to 3 (all of which are needed)...
0xe doesn't look much nicer though. Is there a more Linuxy way to do this?

Reply:
`BIT(3)|BIT(2)|BIT(1)`?